### PR TITLE
BAU - Downgrade Maven failsafe to 3.0.0-M1

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Downgrade Maven failsafe plugin as latest version doesn't run integration tests on `mvn clean verify`
- Fixed few failing integration tests
- Run integration tests on master

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -317,15 +317,10 @@
                     </execution>
                 </executions>
             </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-surefire-plugin</artifactId>
-                <version>${surefire.version}</version>
-            </plugin>
```
Comment (alexbishop1) [inline pom.xml]: Why was this removed?

Comment (kbottla) [reply pom.xml]: We already define surefire-plugin in profiles (default & contract-tests) and this is redundant
